tests/test_measurements.py

```diff
@@ -757,9 +757,9 @@ def circuit():
         res = circuit()
 
         assert isinstance(res, tuple)
-        assert res[0] == {basis_state: shot_vec[0]}
-        assert res[1] == {basis_state: shot_vec[1]}
-        assert res[2] == {basis_state: shot_vec[2]}
+        assert res[0][basis_state] == shot_vec[0]
```

Comment:
I've reverted these back to how they were previously, so that they just test the default implementation of `counts`. I've added new functions to test `counts(all_outcomes=True)`.